packages/access-api/test/store-list.js

```diff
@@ -0,0 +1,197 @@
+import * as ed25519 from '@ucanto/principal/ed25519'
+import * as dagUcanDid from '@ipld/dag-ucan/did'
+import { isUploadApiStack } from '../src/service/upload-api-proxy.js'
+
+describe('proxy store/list invocations to upload-api', function () {
+  it('forwards store/list invocations with aud=did:key', async function () {
+    const { issuer, service, conn } = await context({
+      environment: {
+        ...process.env,
+        PRIVATE_KEY:
+          process.env.WEB3_STORAGE_PRIVATE_KEY || process.env.PRIVATE_KEY,
+      },
+    })
+    const spaceCreation = await createSpace(
+      issuer,
+      service,
+      conn,
+      'space-info@dag.house'
+    )
+    const listInvocation = Store.list.invoke({
+      issuer,
+      audience: service,
+      proofs: [spaceCreation.delegation],
+      with: spaceCreation.space.did(),
+      nb: {},
+    })
+    const result = await listInvocation.execute(
+      /** @type {import('@ucanto/interface').ConnectionView<any>} */ (conn)
+    )
+    if (result?.error) {
+      try {
+        /**
+         * This is expected to error even when the forwarding is working.
+         * When this happens, the error will be like
+         * InvalidAudience: Delegation audience is 'did:key:z6MkqBzPG7oNu7At8fktasQuS7QR7Tj7CujaijPMAgzdmAxD' instead of 'did:web:web3.storage'\n"
+         * This makes sense because this test is just ensuring that the invocation is forwarded based on the capability.can value
+         * even though the audience is just some random did:key
+         */
+        assert.ok('name' in result, 'result has name property')
+        assert.strictEqual(
+          result.name,
+          'InvalidAudience',
+          'error has expected name'
+        )
+        assert.strictEqual(
+          result.message?.includes(
+            `Delegation audience is '${service.did()}' instead of 'did:web:web3.storage'`
+          ),
+          true,
+          `error includes expected message`
+        )
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('unexpected error', result)
+        throw error
+      }
+    }
+  })
+  it('forwards invocations with aud=did:web:web3.storage', async function () {
+    const web3storageDid = 'did:web:web3.storage'
+    // if this is set, it's to inject in the actual private key used by web3StorageDid.
+    // and if it's present, the assertions will expect no error from the proxy or upstream
+    const privateKeyFromEnv = process.env.WEB3_STORAGE_PRIVATE_KEY
+    const {
+      issuer,
+      service: serviceSigner,
+      conn,
+    } = await context({
+      environment: {
+        ...process.env,
+        // this emulates the configuration for deployed environments,
+        // which will allow the access-api ucanto server to accept
+        // invocations where aud=web3storageDid
+        DID: web3storageDid,
+        PRIVATE_KEY: privateKeyFromEnv ?? process.env.PRIVATE_KEY,
```

Comment:
some of this extra stuff is in here because I wanted to support running the tests with the production private key, which works with a non-error result. But if you don't pass that in, the assertions shift slightly to ensuring that it's an error result of the kind that would happen only if the server signer is wrong

Note how this test was run with `WEB3_STORAGE_PRIVATE_KEY`:
```
bengo@bengo ~/protocol.ai/w3protocol/packages/access-api ⚡  DID="$PROD_DID_KEY" WEB3_STORAGE_PRIVATE_KEY="$PROD_PRIVATE_KEY" pnpm test -- -f 'upload-api'

> @web3-storage/access-api@4.2.0 test /Users/bengo/protocol.ai/w3protocol/packages/access-api
> pnpm build && mocha --bail --timeout 10s -n no-warnings -n experimental-vm-modules "-f" "upload-api"


> @web3-storage/access-api@4.2.0 build /Users/bengo/protocol.ai/w3protocol/packages/access-api
> scripts/cli.js build



  proxy store/list invocations to upload-api
    ✔ forwards store/list invocations with aud=did:key (648ms)
    ✔ forwards invocations with aud=did:web:web3.storage (460ms)
    ✔ errors when a bad delegation is given as proof (320ms)

  Store.all
    ✔ proxies store/add to upload-api (331ms)
    ✔ proxies store/remove to upload-api (306ms)
    ✔ proxies store/list to upload-api (385ms)

  Upload.all
    ✔ proxies upload/add to upload-api (303ms)
    ✔ proxies upload/remove to upload-api (287ms)
    ✔ proxies upload/list to upload-api (342ms)


  9 passing (3s)